base/loading.jl

```diff
@@ -1657,14 +1664,11 @@ const COMPILETIME_PREFERENCES = Dict{UUID,Set{String}}()
 # we mark that usage as a usage at compile-time and call this method, so that at the end of `.ji` generation,
 # we can record the list of compile-time preferences and embed that into the `.ji` header
 function record_compiletime_preference(uuid::UUID, key::String)
-    if !haskey(COMPILETIME_PREFERENCES, uuid)
-        COMPILETIME_PREFERENCES[uuid] = Set((key,))
-    else
-        push!(COMPILETIME_PREFERENCES[uuid], key)
-    end
+    pref = get!(Set{String}, COMPILETIME_PREFERENCES, uuid)
+    push!(pref, key)
     return nothing
 end
-get_compiletime_preferences(uuid::UUID) = collect(get(COMPILETIME_PREFERENCES, uuid, String[]))
+get_compiletime_preferences(uuid::UUID) = collect(get(Vector{String}, COMPILETIME_PREFERENCES, uuid))
 get_compiletime_preferences(m::Module) = get_compiletime_preferences(PkgId(m).uuid)
 get_compiletime_preferences(::Nothing) = String[]
```

Comment:
We don't use `nothing` as a sentinel value anymore, so you can drop the `::Nothing` method here as well.

Reply:
Doesn't look like it: https://build.julialang.org/#/builders/20/builds/5814/steps/7/logs/stdio. It happens when `PkgId(m)` doesn't identify a uuid, just a name.

Reply:
Ah, good call.  Thank God for tests!